ports/stm32f4/common-hal/analogio/AnalogOut.c

```diff
@@ -40,24 +40,27 @@
 #include "stm32f4xx_hal.h"
 
 //DAC is shared between both channels. 
-//TODO: store as struct with channel info, automatically turn it off if unused 
-//on both channels for power save?
 #if HAS_DAC
 DAC_HandleTypeDef handle;
 #endif
 
+STATIC bool dac_on[2];
+
 void common_hal_analogio_analogout_construct(analogio_analogout_obj_t* self,
         const mcu_pin_obj_t *pin) {
     #if !(HAS_DAC)
     mp_raise_ValueError(translate("No DAC on chip"));
     #else
     if (pin == &pin_PA04) {
         self->channel = DAC_CHANNEL_1;
+        self->dac_index = 0;
     } else if (pin == &pin_PA05) {
         self->channel = DAC_CHANNEL_2;
+        self->dac_index = 1;
     } else {
         mp_raise_ValueError(translate("Invalid DAC pin supplied"));
     }
+    dac_on[self->dac_index] = true;
```

Comment:
Move this down below the initialization to line 89 or so, because if you throw an exception, `dac_on` will incorrectly remain set.